app/views/devise/mailer/confirmation_instructions.html.erb

```diff
@@ -1,5 +1,6 @@
-<p>Welcome <%= @email %>!</p>
+<% require 'devise/version' %>
+<p><%= t('.greeting', :recipient => @resource.email, :default => "Welcome #{@resource.email}!") %></p>
 
-<p>You can confirm your account email through the link below:</p>
-
-<p><%= link_to 'Confirm my account', confirmation_url(@resource, :confirmation_token => @token) %></p>
+<p><%= t('.instruction', :default => "You can confirm your account email through the link below:") %></p>
+<p><%= link_to t('.action', :default => "Confirm my account"),
+         confirmation_url(@resource, :confirmation_token => (Devise::VERSION.start_with?('3.') ? @token : @resource.confirmation_token)) %></p>
```

Comment:
Isn't there a better way to do this? We need to use "@resource.confirmation_token" with versions 2.* or 1.*? What will happen when version 4.* is released?